src/App.tsx

```diff
@@ -1,27 +1,60 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
 // or
 // import * as goodsAPI from './api/goods';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goodsToShow, setGoodsToShow] = useState<Good[]>([]);
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  useEffect(() => {
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+  }, []);
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+  const loadAllGoods = () => getAll()
+    .then((goods) => setGoodsToShow([...goods]));
 
-    <GoodsList goods={[]} />
-  </div>
-);
+  const loadFiveFirstGoods = () => get5First()
+    .then((goods) => setGoodsToShow([...goods]));
+
+  const loadRedGoods = () => getRedGoods()
+    .then((goods) => setGoodsToShow([...goods]));
```

Comment:
You need to:
- show loader when fetching data;
- catch and show errors;
- prevent clicking same button twice (don't load same data multiple times in a row)

Reply:
all done